File: java/app/src/main/java/githubtododemo/githubclient/models/NullableMilestone.java
package githubtododemo.githubclient.models;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.time.OffsetDateTime;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
/**
 * A collection of related issues and pull requests.
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class NullableMilestone implements Parsable {
    /**
     * The closed_at property
     */
    private OffsetDateTime closedAt;
    /**
     * The closed_issues property
     */
    private Integer closedIssues;
    /**
     * The created_at property
     */
    private OffsetDateTime createdAt;
    /**
     * A GitHub user.
     */
    private NullableSimpleUser creator;
    /**
     * The description property
     */
    private String description;
    /**
     * The due_on property
     */
    private OffsetDateTime dueOn;
    /**
     * The html_url property
     */
    private String htmlUrl;
    /**
     * The id property
     */
    private Integer id;
    /**
     * The labels_url property
     */
    private String labelsUrl;
    /**
     * The node_id property
     */
    private String nodeId;
    /**
     * The number of the milestone.
     */
    private Integer number;
    /**
     * The open_issues property
     */
    private Integer openIssues;
    /**
     * The state of the milestone.
     */
    private NullableMilestoneState state;
    /**
     * The title of the milestone.
     */
    private String title;
    /**
     * The updated_at property
     */
    private OffsetDateTime updatedAt;
    /**
     * The url property
     */
    private String url;
    /**
     * Instantiates a new {@link NullableMilestone} and sets the default values.
     */
    public NullableMilestone() {
        this.setState(NullableMilestoneState.forValue("open"));
    }
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param parseNode The parse node to use to read the discriminator value and create the object
     * @return a {@link NullableMilestone}
     */
    @jakarta.annotation.Nonnull
    public static NullableMilestone createFromDiscriminatorValue(@jakarta.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new NullableMilestone();
    }
    /**
     * Gets the closed_at property value. The closed_at property
     * @return a {@link OffsetDateTime}
     */
    @jakarta.annotation.Nullable
    public OffsetDateTime getClosedAt() {
        return this.closedAt;
    }
    /**
     * Gets the closed_issues property value. The closed_issues property
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getClosedIssues() {
        return this.closedIssues;
    }
    /**
     * Gets the created_at property value. The created_at property
     * @return a {@link OffsetDateTime}
     */
    @jakarta.annotation.Nullable
    public OffsetDateTime getCreatedAt() {
        return this.createdAt;
    }
    /**
     * Gets the creator property value. A GitHub user.
     * @return a {@link NullableSimpleUser}
     */
    @jakarta.annotation.Nullable
    public NullableSimpleUser getCreator() {
        return this.creator;
    }
    /**
     * Gets the description property value. The description property
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getDescription() {
        return this.description;
    }
    /**
     * Gets the due_on property value. The due_on property
     * @return a {@link OffsetDateTime}
     */
    @jakarta.annotation.Nullable
    public OffsetDateTime getDueOn() {
        return this.dueOn;
    }
    /**
     * The deserialization information for the current model
     * @return a {@link Map<String, java.util.function.Consumer<ParseNode>>}
     */
    @jakarta.annotation.Nonnull
    public Map<String, java.util.function.Consumer<ParseNode>> getFieldDeserializers() {
        final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap = new HashMap<String, java.util.function.Consumer<ParseNode>>(16);
        deserializerMap.put("closed_at", (n) -> { this.setClosedAt(n.getOffsetDateTimeValue()); });
        deserializerMap.put("closed_issues", (n) -> { this.setClosedIssues(n.getIntegerValue()); });
        deserializerMap.put("created_at", (n) -> { this.setCreatedAt(n.getOffsetDateTimeValue()); });
        deserializerMap.put("creator", (n) -> { this.setCreator(n.getObjectValue(NullableSimpleUser::createFromDiscriminatorValue)); });
        deserializerMap.put("description", (n) -> { this.setDescription(n.getStringValue()); });
        deserializerMap.put("due_on", (n) -> { this.setDueOn(n.getOffsetDateTimeValue()); });
        deserializerMap.put("html_url", (n) -> { this.setHtmlUrl(n.getStringValue()); });
        deserializerMap.put("id", (n) -> { this.setId(n.getIntegerValue()); });
        deserializerMap.put("labels_url", (n) -> { this.setLabelsUrl(n.getStringValue()); });
        deserializerMap.put("node_id", (n) -> { this.setNodeId(n.getStringValue()); });
        deserializerMap.put("number", (n) -> { this.setNumber(n.getIntegerValue()); });
        deserializerMap.put("open_issues", (n) -> { this.setOpenIssues(n.getIntegerValue()); });
        deserializerMap.put("state", (n) -> { this.setState(n.getEnumValue(NullableMilestoneState::forValue)); });
        deserializerMap.put("title", (n) -> { this.setTitle(n.getStringValue()); });
        deserializerMap.put("updated_at", (n) -> { this.setUpdatedAt(n.getOffsetDateTimeValue()); });
        deserializerMap.put("url", (n) -> { this.setUrl(n.getStringValue()); });
        return deserializerMap;
    }
    /**
     * Gets the html_url property value. The html_url property
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getHtmlUrl() {
        return this.htmlUrl;
    }
    /**
     * Gets the id property value. The id property
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getId() {
        return this.id;
    }
    /**
     * Gets the labels_url property value. The labels_url property
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getLabelsUrl() {
        return this.labelsUrl;
    }
    /**
     * Gets the node_id property value. The node_id property
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getNodeId() {
        return this.nodeId;
    }
    /**
     * Gets the number property value. The number of the milestone.
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getNumber() {
        return this.number;
    }
    /**
     * Gets the open_issues property value. The open_issues property
     * @return a {@link Integer}
     */
    @jakarta.annotation.Nullable
    public Integer getOpenIssues() {
        return this.openIssues;
    }
    /**
     * Gets the state property value. The state of the milestone.
     * @return a {@link NullableMilestoneState}
     */
    @jakarta.annotation.Nullable
    public NullableMilestoneState getState() {
        return this.state;
    }
    /**
     * Gets the title property value. The title of the milestone.
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getTitle() {
        return this.title;
    }
    /**
     * Gets the updated_at property value. The updated_at property
     * @return a {@link OffsetDateTime}
     */
    @jakarta.annotation.Nullable
    public OffsetDateTime getUpdatedAt() {
        return this.updatedAt;
    }
    /**
     * Gets the url property value. The url property
     * @return a {@link String}
     */
    @jakarta.annotation.Nullable
    public String getUrl() {
        return this.url;
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public void serialize(@jakarta.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeOffsetDateTimeValue("closed_at", this.getClosedAt());
        writer.writeIntegerValue("closed_issues", this.getClosedIssues());
        writer.writeOffsetDateTimeValue("created_at", this.getCreatedAt());
        writer.writeObjectValue("creator", this.getCreator());
        writer.writeStringValue("description", this.getDescription());
        writer.writeOffsetDateTimeValue("due_on", this.getDueOn());
        writer.writeStringValue("html_url", this.getHtmlUrl());
        writer.writeIntegerValue("id", this.getId());
        writer.writeStringValue("labels_url", this.getLabelsUrl());
        writer.writeStringValue("node_id", this.getNodeId());
        writer.writeIntegerValue("number", this.getNumber());
        writer.writeIntegerValue("open_issues", this.getOpenIssues());
        writer.writeEnumValue("state", this.getState());
        writer.writeStringValue("title", this.getTitle());
        writer.writeOffsetDateTimeValue("updated_at", this.getUpdatedAt());
        writer.writeStringValue("url", this.getUrl());
    }
    /**
     * Sets the closed_at property value. The closed_at property
     * @param value Value to set for the closed_at property.
     */
    public void setClosedAt(@jakarta.annotation.Nullable final OffsetDateTime value) {
        this.closedAt = value;
    }
    /**
     * Sets the closed_issues property value. The closed_issues property
     * @param value Value to set for the closed_issues property.
     */
    public void setClosedIssues(@jakarta.annotation.Nullable final Integer value) {
        this.closedIssues = value;
    }
    /**
     * Sets the created_at property value. The created_at property
     * @param value Value to set for the created_at property.
     */
    public void setCreatedAt(@jakarta.annotation.Nullable final OffsetDateTime value) {
        this.createdAt = value;
    }
    /**
     * Sets the creator property value. A GitHub user.
     * @param value Value to set for the creator property.
     */
    public void setCreator(@jakarta.annotation.Nullable final NullableSimpleUser value) {
        this.creator = value;
    }
    /**
     * Sets the description property value. The description property
     * @param value Value to set for the description property.
     */
    public void setDescription(@jakarta.annotation.Nullable final String value) {
        this.description = value;
    }
    /**
     * Sets the due_on property value. The due_on property
     * @param value Value to set for the due_on property.
     */
    public void setDueOn(@jakarta.annotation.Nullable final OffsetDateTime value) {
        this.dueOn = value;
    }
    /**
     * Sets the html_url property value. The html_url property
     * @param value Value to set for the html_url property.
     */
    public void setHtmlUrl(@jakarta.annotation.Nullable final String value) {
        this.htmlUrl = value;
    }
    /**
     * Sets the id property value. The id property
     * @param value Value to set for the id property.
     */
    public void setId(@jakarta.annotation.Nullable final Integer value) {
        this.id = value;
    }
    /**
     * Sets the labels_url property value. The labels_url property
     * @param value Value to set for the labels_url property.
     */
    public void setLabelsUrl(@jakarta.annotation.Nullable final String value) {
        this.labelsUrl = value;
    }
    /**
     * Sets the node_id property value. The node_id property
     * @param value Value to set for the node_id property.
     */
    public void setNodeId(@jakarta.annotation.Nullable final String value) {
        this.nodeId = value;
    }
    /**
     * Sets the number property value. The number of the milestone.
     * @param value Value to set for the number property.
     */
    public void setNumber(@jakarta.annotation.Nullable final Integer value) {
        this.number = value;
    }
    /**
     * Sets the open_issues property value. The open_issues property
     * @param value Value to set for the open_issues property.
     */
    public void setOpenIssues(@jakarta.annotation.Nullable final Integer value) {
        this.openIssues = value;
    }
    /**
     * Sets the state property value. The state of the milestone.
     * @param value Value to set for the state property.
     */
    public void setState(@jakarta.annotation.Nullable final NullableMilestoneState value) {
        this.state = value;
    }
    /**
     * Sets the title property value. The title of the milestone.
     * @param value Value to set for the title property.
     */
    public void setTitle(@jakarta.annotation.Nullable final String value) {
        this.title = value;
    }
    /**
     * Sets the updated_at property value. The updated_at property
     * @param value Value to set for the updated_at property.
     */
    public void setUpdatedAt(@jakarta.annotation.Nullable final OffsetDateTime value) {
        this.updatedAt = value;
    }
    /**
     * Sets the url property value. The url property
     * @param value Value to set for the url property.
     */
    public void setUrl(@jakarta.annotation.Nullable final String value) {
        this.url = value;
    }
}

File: python/githubclient/models/nullable_milestone.py
from __future__ import annotations
import datetime
from dataclasses import dataclass, field
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .nullable_milestone_state import NullableMilestone_state
    from .nullable_simple_user import NullableSimpleUser


@dataclass
class NullableMilestone(AdditionalDataHolder, Parsable):
    """
    A collection of related issues and pull requests.
    """
    # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additional_data: Dict[str, Any] = field(default_factory=dict)

    from .nullable_milestone_state import NullableMilestone_state

    # The state of the milestone.
    state: Optional[NullableMilestone_state] = NullableMilestone_state("open")
    # The closed_at property
    closed_at: Optional[datetime.datetime] = None
    # The closed_issues property
    closed_issues: Optional[int] = None
    # The created_at property
    created_at: Optional[datetime.datetime] = None
    # A GitHub user.
    creator: Optional[NullableSimpleUser] = None
    # The description property
    description: Optional[str] = None
    # The due_on property
    due_on: Optional[datetime.datetime] = None
    # The html_url property
    html_url: Optional[str] = None
    # The id property
    id: Optional[int] = None
    # The labels_url property
    labels_url: Optional[str] = None
    # The node_id property
    node_id: Optional[str] = None
    # The number of the milestone.
    number: Optional[int] = None
    # The open_issues property
    open_issues: Optional[int] = None
    # The title of the milestone.
    title: Optional[str] = None
    # The updated_at property
    updated_at: Optional[datetime.datetime] = None
    # The url property
    url: Optional[str] = None

    @staticmethod
    def create_from_discriminator_value(
            parse_node: Optional[ParseNode] = None) -> NullableMilestone:
        """
        Creates a new instance of the appropriate class based on discriminator value
        param parse_node: The parse node to use to read the discriminator value and create the object
        Returns: NullableMilestone
        """
        if not parse_node:
            raise TypeError("parse_node cannot be null.")
        return NullableMilestone()

    def get_field_deserializers(
        self, ) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        from .nullable_milestone_state import NullableMilestone_state
        from .nullable_simple_user import NullableSimpleUser

        from .nullable_milestone_state import NullableMilestone_state
        from .nullable_simple_user import NullableSimpleUser

        fields: Dict[str, Callable[[Any], None]] = {
            "closed_at":
            lambda n: setattr(self, 'closed_at', n.get_datetime_value()),
            "closed_issues":
            lambda n: setattr(self, 'closed_issues', n.get_int_value()),
            "created_at":
            lambda n: setattr(self, 'created_at', n.get_datetime_value()),
            "creator":
            lambda n: setattr(self, 'creator',
                              n.get_object_value(NullableSimpleUser)),
            "description":
            lambda n: setattr(self, 'description', n.get_str_value()),
            "due_on":
            lambda n: setattr(self, 'due_on', n.get_datetime_value()),
            "html_url":
            lambda n: setattr(self, 'html_url', n.get_str_value()),
            "id":
            lambda n: setattr(self, 'id', n.get_int_value()),
            "labels_url":
            lambda n: setattr(self, 'labels_url', n.get_str_value()),
            "node_id":
            lambda n: setattr(self, 'node_id', n.get_str_value()),
            "number":
            lambda n: setattr(self, 'number', n.get_int_value()),
            "open_issues":
            lambda n: setattr(self, 'open_issues', n.get_int_value()),
            "state":
            lambda n: setattr(self, 'state',
                              n.get_enum_value(NullableMilestone_state)),
            "title":
            lambda n: setattr(self, 'title', n.get_str_value()),
            "updated_at":
            lambda n: setattr(self, 'updated_at', n.get_datetime_value()),
            "url":
            lambda n: setattr(self, 'url', n.get_str_value()),
        }
        return fields

    def serialize(self, writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        param writer: Serialization writer to use to serialize this model
        Returns: None
        """
        if not writer:
            raise TypeError("writer cannot be null.")
        writer.write_datetime_value("closed_at", self.closed_at)
        writer.write_int_value("closed_issues", self.closed_issues)
        writer.write_datetime_value("created_at", self.created_at)
        writer.write_object_value("creator", self.creator)
        writer.write_str_value("description", self.description)
        writer.write_datetime_value("due_on", self.due_on)
        writer.write_str_value("html_url", self.html_url)
        writer.write_int_value("id", self.id)
        writer.write_str_value("labels_url", self.labels_url)
        writer.write_str_value("node_id", self.node_id)
        writer.write_int_value("number", self.number)
        writer.write_int_value("open_issues", self.open_issues)
        writer.write_enum_value("state", self.state)
        writer.write_str_value("title", self.title)
        writer.write_datetime_value("updated_at", self.updated_at)
        writer.write_str_value("url", self.url)
        writer.write_additional_data_value(self.additional_data)
